edge/cmd/edgecore/app/server.go

```diff
@@ -82,12 +85,19 @@ offering HTTP client capabilities to components of cloud to reach HTTP servers r
 				}
 			}
 
-			// get edge node local ip
-			if config.Modules.Edged.NodeIP == "" {
-				hostnameOverride := util.GetHostname()
-				localIP, _ := util.GetLocalIP(hostnameOverride)
+			// Get edge node local ip only when the custiomInterfaceName has been set.
+			// Defaults to the local IP from the default interface by the default config
+			if config.Modules.Edged.CustomInterfaceName != constants.DefaultCustomInterfaceName {
+				var ip net.IP
+				var localIP string
+				klog.Infof("Getting IP address by custom interface: %s", config.Modules.Edged.CustomInterfaceName)
+				ip, err = netutil.ChooseBindAddressForInterface(config.Modules.Edged.CustomInterfaceName)
+				if ip != nil {
+					localIP = ip.String()
+				}
 				config.Modules.Edged.NodeIP = localIP
```

Comment:
Is is possible to set nodeIP directly instead of get it through inerface name?

Reply:
I explicitly want to use the interface name. However it is still possible to manually set the NodeIP, if the `CustomInterfaceName` is empty in the config.

Reply:
I think that feature can be added in the feature with separate pr

Reply:
Oh I think I get it now

Reply:
@fisherxu you can set the IP Address by hand in the configuration file already. Do you mean that?

Reply:
Yes, because I noticed in k8s/kubelet there's no interface name flag, only has nodeIP. I have another question, if users specify both nodeIP and interfaceName, seems now the nodeIP will be overided, we can add some comments that interfaceName has a higher priority if both specified.

Reply:
Already has comments now :)